### PR TITLE
UI for Overrides

### PR DESCRIPTION
**Pull request by tvrzna**
References #31.

In Repository view page in was added menu item Overrides. In first visit it creates folder for overrides and allows to list it.

In every listed folder there is possibility to create new folder, create new empty file or upload files into current folder. Every file and folder could be deleted and renamed, for files there are two more options: first is simple text edit, it means it won't show images, but allows to change e.g. build script, README, ... Second is download the file.

As I already mentioned in issue, previous commits were rebased so it also contains latest changes from master.

In this state it is far from being perfect, but I believe that further development requires insight from more people.

### CODE REVIEW

File: flux/views.py [hunk 1]
```diff
@@ -523,6 +525,207 @@ def ping_repo():
...
+  path = request.args.get('path', '')
+
+  repo = Repository.get(id=repo_id)
+  if not repo:
+    return abort(404)
+
+  if action == OVERRIDES_ACTION_CREATEFOLDER:
+    name = request.args.get('name', '').replace('../', '')
+    try:
+      file_utils.create_folder(os.path.join(utils.get_override_path(repo), path.replace('/', os.sep)), name)
```
Comment (gsantner) [inline flux/views.py]: looking at this all - how is it with non-unix OS?

Comment (tvrzna) [reply flux/views.py]: Line 697 is simple protection from users trying to create folder in other location.
Line 699 uses same folder creating as build.py.

But I can't guarantee it works on non-unix OS correctly, I did not test it elsewhere.

Comment (gsantner) [reply flux/views.py]: no i didnt mean the specific line but all lines nearby

Comment (tvrzna) [reply flux/views.py]: Oh, I'm sorry, I thought you mean only lines 696-699.

Separator is used on URL path from browser, so format is not OS specific.

Comment (gsantner) [reply flux/views.py]: oh, this is all about web content :D

**Comment by NiklasRosenstein** [comment]
Please avoid plain `except:` clauses. `os.mknod()` does not exist on Windows, and that error was caught and ignored. I've added some logging for now, at least.

**Comment by NiklasRosenstein** [comment]
The path sanitization needs to be a bit more sophisticated, I can for example use `..\foo.txt` on Windows to create a file outside the designated override directory. But the rest looks good! :)

**Comment by tvrzna** [comment]
Thank you for review and fixes :)
When I get back to my computer, I'll use `secure_filename` method from flask (as it is used for uploaded files), it should be good enough.

**Comment by NiklasRosenstein** [comment]
I was going to suggest that function, but it is for filenames not paths, so any slashes will be converted to underscores. What you could do instead is to construct the absolute path, and then check if it is inside the allowed directory.

**Comment by tvrzna** [comment]
Before file and folder are created, there is just their "new" name and it could be sanitized by that function.

**Comment by gsantner** [comment]
is it ready to merge? :D

**Comment by NiklasRosenstein** [comment]
Sorry for the delay, yes I think this is fine now. :)

**Pull request merged**